pkg/terraform/v1beta2/config.go

```diff
@@ -263,6 +265,20 @@ func (output *Config) Apply(cluster *kubeonev1beta2.KubeOneCluster) error {
 		cluster.CloudProvider.Hetzner.NetworkID = cp.NetworkID
 	}
 
+	if cluster.CloudProvider.VMwareCloudDirector != nil {
+		// VAppName is used only for VMware Cloud Director.
+		if len(cp.VAppName) > 0 {
+			cluster.CloudProvider.VMwareCloudDirector.VApp = cp.VAppName
+		} else {
+			cluster.CloudProvider.VMwareCloudDirector.VApp = cluster.Name
```

Comment:
This logic should be part of the defaulting process.

Reply:
Umm, are you referring to https://github.com/kubermatic/kubeone/blob/master/pkg/apis/kubeone/v1beta2/defaults.go#L47? This defaulting here is based on the terraform output. Just like https://github.com/kubermatic/kubeone/blob/master/pkg/terraform/v1beta2/config.go#L263